maltoolbox/language/languagegraph.py

```diff
@@ -411,39 +668,241 @@ def from_mar_archive(cls, mar_archive: str) -> LanguageGraph:
+                            target_full_attack_step_name)
+                    target_asset = lang_graph.assets[target_asset_name]
+                    target_attack_step = target_asset.attack_steps[
+                        target_attack_step_name]
+                    for dep_chain_dict in dep_chains:
+                        dep_chain = DependencyChain._from_dict(
+                            dep_chain_dict,
+                            lang_graph
+                        )
+                        if target_attack_step.full_name in attack_step.children:
+                            attack_step.children[target_attack_step.full_name].\
+                                append((target_attack_step, dep_chain))
+                        else:
+                            attack_step.children[target_attack_step.full_name] = \
+                                [(target_attack_step, dep_chain)]
+
+                for parent_target in attack_step_dict['parents'].items():
+                    target_full_attack_step_name = parent_target[0]
+                    dep_chains = parent_target[1]
+                    target_asset_name = get_asset_name_from_full_attack_step(
+                        target_full_attack_step_name)
+                    target_attack_step_name = \
+                        get_attack_step_name_from_full_attack_step(
+                            target_full_attack_step_name)
+                    target_asset = lang_graph.assets[target_asset_name]
+                    target_attack_step = target_asset.attack_steps[
+                        target_attack_step_name]
+                    for dep_chain_dict in dep_chains:
+                        dep_chain = DependencyChain._from_dict(
+                            dep_chain_dict,
+                            lang_graph
+                        )
+                        if target_attack_step.full_name in attack_step.parents:
+                            attack_step.parents[target_attack_step.full_name].\
+                                append((target_attack_step, dep_chain))
+                        else:
+                            attack_step.parents[target_attack_step.full_name] = \
+                                [(target_attack_step, dep_chain)]
+
+                # Recreate the requirements of exist and notExist attack steps
+                if attack_step.type == 'exist' or \
+                        attack_step.type == 'notExist':
+                    if 'requirements' in attack_step_dict:
+                        dep_chains = attack_step_dict['requirements']
+                        attack_step.requirements = []
+                        for dep_chain_dict in dep_chains:
+                            dep_chain = DependencyChain._from_dict(
+                                dep_chain_dict,
+                                lang_graph
+                            )
+                            attack_step.requirements.append(dep_chain)
```

Comment:
This is not the best result, as it will create a step object that contains all its requirements ready to go with inheritance resolved, while the same step, if not loaded from a file but constructed on the fly from the mal and model files will not. How often is there the need to load an attack graph from a file? Why is this needed instead of constructing the attack graph always? Is it due to performance?